pyproject.toml

```diff
@@ -36,50 +38,53 @@ dynamic = ["version"]
 name = "Frequenz Energy-as-a-Service GmbH"
 email = "floss@frequenz.com"
 
+# TODO(cookiecutter): Remove and add more optional dependencies if appropriate
 [project.optional-dependencies]
 dev-flake8 = [
   "flake8 == 6.1.0",
   "flake8-docstrings == 1.7.0",
   "flake8-pyproject == 1.2.3",  # For reading the flake8 config from pyproject.toml
-  "pydoclint == 0.3.1",
-]
-dev-docstrings = [
+  "pydoclint == 0.3.2",
   "pydocstyle == 6.3.0",
-  "darglint == 1.8.1",
 ]
-dev-formatting = ["black == 23.7.0", "isort == 5.12.0"]
+dev-formatting = ["black == 23.9.1", "isort == 5.12.0"]
 dev-mkdocs = [
+  "black == 23.9.1",
+  "Markdown==3.4.4",
   "mike == 2.0.0",
   "mkdocs-gen-files == 0.5.0",
-  "mkdocs-literate-nav == 0.6.0",
-  "mkdocs-material == 9.4.4",
-  "mkdocstrings[python] == 0.22.0",
-  "frequenz-repo-config[api] == 0.6.2",
+  "mkdocs-literate-nav == 0.6.1",
+  "mkdocs-macros-plugin == 1.0.4",
+  "mkdocs-material == 9.3.1",
```

Comment:
You probably don't want to downgrade `mkdocs-material`, and actually you probably don't want to downgrade **any** dependency.